test/factories/host_related.rb

```diff
@@ -320,6 +320,24 @@ def set_environment_taxonomies(host_or_hostgroup, environment = host_or_hostgrou
       end
     end
 
+    factory :host_for_snapshots_ipv4_dhcp_el7 do
+      name { 'snapshot-ipv4-dhcp-el7' }
+      hostname { name }
+      managed { true }
+      operatingsystem { FactoryBot.build(:for_snapshots_centos_7_0) }
```

Comment:
This looks like the wrong way to build associations if I read
https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#associations
```suggestion
      operatingsystem factory: :for_snapshots_centos_7_0
```

Reply:
Well, that does not work. I know very little about FactoryBot and it took me a while to get this working honestly. Note I am not creating those records, I am building them (not saving them in the db).

I am leaving this as-is feel free to show me a working example.